m2c/arch_mips.py

```diff
@@ -1848,3 +1851,284 @@ def function_return(expr: Expression) -> Dict[Register, Expression]:
             ),
             Register("f1"): SecondF64Half(),
         }
+
+
+class MipseeArch(MipsArch):
+    stack_pointer_reg = Register("sp")
+
+    argument_regs = [
+        Register(r)
+        for r in [
+            "a0",
+            "a1",
+            "a2",
+            "a3",
+            "a4",
+            "a5",
+            "a6",
+            "a7",
+            "f12",
+            "f13",
+            "f14",
+            "f15",
+            "f16",
+            "f17",
+            "f18",
+            "f19",
+        ]
+    ]
+    simple_temp_regs = [
+        Register(r)
+        for r in [
+            "v0",
+            "v1",
+            "t4",
+            "t5",
+            "t6",
+            "t7",
+            "t8",
+            "t9",
+            "f0",
+            "f1",
+            "f2",
+            "f3",
+            "f4",
+            "f5",
+            "f6",
+            "f7",
+            "f8",
+            "f9",
+            "f10",
+            "f11",
+            "f21",
+            "f23",
+            "f25",
+            "f27",
+            "f29",
+            "f31",
+        ]
+    ]
+    temp_regs = (
+        argument_regs
+        + simple_temp_regs
+        + [
+            Register(r)
+            for r in [
+                "at",
+                "hi",
+                "lo",
+                "condition_bit",
+            ]
+        ]
+    )
+    saved_regs = [
+        Register(r)
+        for r in [
+            "s0",
+            "s1",
+            "s2",
+            "s3",
+            "s4",
+            "s5",
+            "s6",
+            "s7",
+            "f20",
+            "f22",
+            "f24",
+            "f26",
+            "f28",
+            "f30",
```

Comment:
I imagine f21-f31 as well?

Reply:
In the n32 abi, which I think we decided is closer to what eabi64 is, the odd ones aren't saved it seems. @AngheloAlf does this sound right to you?

Reply:
Empirically all of f20-f31 seem to be saved: https://decomp.me/scratch/f51s0

Reply:
Indeed, odd registers are used and saved in eabi64.
We probably copy-pasted the n32 convention and forgot to update it.

Reply:
gotcha - thanks both of you for checking. updated

Reply:
Can you remove them from `simple_temp_regs` as well?